shell/edit/monitoring.coreos.com.alertmanagerconfig/types/pagerduty.vue

```diff
@@ -24,35 +31,39 @@ export default {
     }
   },
   data() {
-    this.value['httpConfig'] = this.value.httpConfig || {};
-    this.value['sendResolved'] = typeof this.value.send_resolved === 'boolean' ? this.value.send_resolved : true;
-
-    const integrationMapping = {
-      'Events API v2': 'routingKey',
-      Prometheus:      'serviceKey'
-    };
-
-    const integrationTypeOptions = Object.keys(integrationMapping);
-
     return {
       integrationMapping,
       integrationTypeOptions,
-      integrationType:             this.value.serviceKey ? integrationTypeOptions[1] : integrationTypeOptions[0],
-      initialRoutingKeySecretKey:  this.value.routingKey?.key || '',
-      initialRoutingKeySecretName: this.value.routingKey?.name || '',
-      initialServiceKeySecretKey:  this.value.serviceKey?.key || '',
-      initialServiceKeySecretName: this.value.serviceKey?.name || '',
-      view:                        _VIEW,
-      none:                        '__[[NONE]]__',
+      integrationType: this.value.serviceKey ? integrationTypeOptions[1] : integrationTypeOptions[0],
+      view:            _VIEW,
+      none:            '__[[NONE]]__',
     };
   },
+  computed: {
+    initialRoutingKeySecretKey() {
```

Comment:
If it is not influent to mutate the object `this.value`, why do we need to reference it? Could we not have just our data as an abstracted component without mixing up with other models?